### ISSUE TITLE
accuracy and val_accuracy direction is set to min in multi_output automodel

### ISSUE THREAD

**Issue by Mgmaplus** [keras-team/autokeras#1596]
### Bug Description

Can only speak in regards to image classification. Both classification_accuracy and val_classification_accuracy direction is set to min. When working with multi-output I have classification_head_1_accuracy, val_classification_head_1_accuracy, classification_head_2_accuracy, val_classification_head_2_accuracy all set to min. I was only able to change the direction of one of these with objective = kerastuner.Objective('val_classification_head_2_accuracy', direction='max'). Having all set to min gives me doubts and having three set to min and one set to max doesn't seem right either.

### Bug Reproduction
Code for reproducing the bug:

```
Early_Stopping = tf.keras.callbacks.EarlyStopping(monitor='val_loss', patience=4, mode='min',
                                                      restore_best_weights=True, verbose=1)
input_node = ak.ImageInput()
output_node_1 = ak.ImageBlock(
        block_type="efficient",
        augment=True)(input_node)
output_node_2 = ak.ResNetBlock(pretrained=True, version='v2')(input_node)
output_node = ak.Merge()([output_node_1, output_node_2])
output_node_style = ak.ClassificationHead()(output_node)
output_node_gender = ak.ClassificationHead()(output_node) 

clf = ak.AutoModel(inputs=input_node, outputs=[output_node_style, output_node_gender], max_trials=3,
                       distribution_strategy=tf.distribute.MirroredStrategy(),
                       project_name='multi_output', objective=kt.Objective('val_classification_head_2_accuracy', direction='max'))  
   
clf.fit(x=X, y=[y, z], batch_size=16, epochs=15, validation_split=.15, callbacks=[Early_Stopping])
```

Data used by the code:
X, y, z = data_full, labels, labels_b
idx = list(range(len(X)))
random.shuffle(idx)
ai = np.array(idx)
X = X[ai]
y = y[ai]
z = z[ai]

### Trial.json

{"trial_id": "97d06993388bc05d354282ca7b0d8ea6", "hyperparameters": {"space": [{"class_name": "Boolean", "config": {"name": "image_block_1/normalize", "default": false, "conditions": []}}, {"class_name": "Boolean", "config": {"name": "image_block_1/efficient_net_block_1/pretrained", "default": false, "conditions": []}}, {"class_name": "Choice", "config": {"name": "image_block_1/efficient_net_block_1/version", "default": "b0", "conditions": [], "values": ["b0", "b1", "b2", "b3", "b4", "b5", "b6", "b7"], "ordered": false}}, {"class_name": "Boolean", "config": {"name": "image_block_1/efficient_net_block_1/imagenet_size", "default": false, "conditions": []}}, {"class_name": "Choice", "config": {"name": "conv_block_1/kernel_size", "default": 3, "conditions": [], "values": [3, 5, 7], "ordered": true}}, {"class_name": "Boolean", "config": {"name": "conv_block_1/separable", "default": false, "conditions": []}}, {"class_name": "Boolean", "config": {"name": "conv_block_1/max_pooling", "default": true, "conditions": []}}, {"class_name": "Choice", "config": {"name": "conv_block_1/dropout", "default": 0, "conditions": [], "values": [0.0, 0.25, 0.5], "ordered": true}}, {"class_name": "Choice", "config": {"name": "conv_block_1/num_blocks", "default": 2, "conditions": [], "values": [1, 2, 3], "ordered": true}}, {"class_name": "Choice", "config": {"name": "conv_block_1/num_layers", "default": 2, "conditions": [], "values": [1, 2], "ordered": true}}, {"class_name": "Choice", "config": {"name": "conv_block_1/filters_0_0", "default": 32, "conditions": [], "values": [16, 32, 64, 128, 256, 512], "ordered": true}}, {"class_name": "Choice", "config": {"name": "conv_block_1/filters_0_1", "default": 32, "conditions": [], "values": [16, 32, 64, 128, 256, 512], "ordered": true}}, {"class_name": "Choice", "config": {"name": "conv_block_1/filters_1_0", "default": 32, "conditions": [], "values": [16, 32, 64, 128, 256, 512], "ordered": true}}, {"class_name": "Choice", "config": {"name": "conv_block_1/filters_1_1", "default": 32, "conditions": [], "values": [16, 32, 64, 128, 256, 512], "ordered": true}}, {"class_name": "Boolean", "config": {"name": "xception_block_1/trainable", "default": false, "conditions": []}}, {"class_name": "Boolean", "config": {"name": "xception_block_1/imagenet_size", "default": false, "conditions": []}}, {"class_name": "Choice", "config": {"name": "classification_head_1/dropout", "default": 0, "conditions": [], "values": [0.0, 0.25, 0.5], "ordered": true}}, {"class_name": "Choice", "config": {"name": "classification_head_2/dropout", "default": 0, "conditions": [], "values": [0.0, 0.25, 0.5], "ordered": true}}, {"class_name": "Choice", "config": {"name": "optimizer", "default": "adam", "conditions": [], "values": ["adam", "sgd", "adam_weight_decay"], "ordered": false}}, {"class_name": "Choice", "config": {"name": "learning_rate", "default": 0.001, "conditions": [], "values": [0.1, 0.01, 0.001, 0.0001, 2e-05, 1e-05], "ordered": true}}, {"class_name": "Choice", "config": {"name": "image_augmentation_1/rotation_factor", "default": 0.0, "conditions": [], "values": [0.0, 0.1], "ordered": true}}, {"class_name": "Choice", "config": {"name": "image_augmentation_1/contrast_factor", "default": 0.0, "conditions": [], "values": [0.0, 0.1], "ordered": true}}], "values": {"image_block_1/normalize": false, "image_block_1/efficient_net_block_1/pretrained": false, "image_block_1/efficient_net_block_1/version": "b0", "image_block_1/efficient_net_block_1/imagenet_size": false, "conv_block_1/kernel_size": 3, "conv_block_1/separable": true, "conv_block_1/max_pooling": true, "conv_block_1/dropout": 0, "conv_block_1/num_blocks": 2, "conv_block_1/num_layers": 2, "conv_block_1/filters_0_0": 32, "conv_block_1/filters_0_1": 32, "conv_block_1/filters_1_0": 32, "conv_block_1/filters_1_1": 32, "xception_block_1/trainable": false, "xception_block_1/imagenet_size": false, "classification_head_1/dropout": 0, "classification_head_2/dropout": 0, "optimizer": "adam", "learning_rate": 0.001, "image_augmentation_1/rotation_factor": 0.0, "image_augmentation_1/contrast_factor": 0.1}}, "metrics": {"metrics": {"loss": {"direction": "min", "observations": [{"value": [24.686279296875], "step": 0}, {"value": [24.71465301513672], "step": 1}, {"value": [24.015743255615234], "step": 2}, {"value": [23.696630477905273], "step": 3}, {"value": [23.104286193847656], "step": 4}, {"value": [23.222652435302734], "step": 5}, {"value": [22.244661331176758], "step": 6}]}, "classification_head_1_loss": {"direction": "min", "observations": [{"value": [5.398770332336426], "step": 0}, {"value": [5.501395225524902], "step": 1}, {"value": [5.668892860412598], "step": 2}, {"value": [5.774673938751221], "step": 3}, {"value": [5.741790294647217], "step": 4}, {"value": [5.870977401733398], "step": 5}, {"value": [5.739967346191406], "step": 6}]}, "classification_head_2_loss": {"direction": "min", "observations": [{"value": [19.287490844726562], "step": 0}, {"value": [19.213272094726562], "step": 1}, {"value": [18.346847534179688], "step": 2}, {"value": [17.921966552734375], "step": 3}, {"value": [17.362497329711914], "step": 4}, {"value": [17.35165786743164], "step": 5}, {"value": [16.50468635559082], "step": 6}]}, **"classification_head_1_accuracy": {"direction": "min"**, "observations": [{"value": [0.6957845687866211], "step": 0}, {"value": [0.7252049446105957], "step": 1}, {"value": [0.7344555258750916], "step": 2}, {"value": [0.7436768412590027], "step": 3}, {"value": [0.7461650967597961], "step": 4}, {"value": [0.7514344453811646], "step": 5}, {"value": [0.7513758540153503], "step": 6}]}, **"classification_head_2_accuracy": {"direction": "min"**, "observations": [{"value": [0.41911593079566956], "step": 0}, {"value": [0.5068208575248718], "step": 1}, {"value": [0.5449941158294678], "step": 2}, {"value": [0.5710480213165283], "step": 3}, {"value": [0.5884367823600769], "step": 4}, {"value": [0.5966627597808838], "step": 5}, {"value": [0.6161592602729797], "step": 6}]}, "val_loss": {"direction": "min", "observations": [{"value": [32.370723724365234], "step": 0}, {"value": [23.471988677978516], "step": 1}, {"value": [27.93853759765625], "step": 2}, {"value": [28.992767333984375], "step": 3}, {"value": [28.89875030517578], "step": 4}, {"value": [29.7956485748291], "step": 5}, {"value": [30.29076385498047], "step": 6}]}, "val_classification_head_1_loss": {"direction": "min", "observations": [{"value": [8.778715133666992], "step": 0}, {"value": [5.744773864746094], "step": 1}, {"value": [6.455005168914795], "step": 2}, {"value": [7.335355281829834], "step": 3}, {"value": [6.837545394897461], "step": 4}, {"value": [6.101838111877441], "step": 5}, {"value": [6.852776050567627], "step": 6}]}, "val_classification_head_2_loss": {"direction": "min", "observations": [{"value": [23.592012405395508], "step": 0}, {"value": [17.727214813232422], "step": 1}, {"value": [21.483531951904297], "step": 2}, {"value": [21.65741539001465], "step": 3}, {"value": [22.061203002929688], "step": 4}, {"value": [23.693805694580078], "step": 5}, {"value": [23.43798828125], "step": 6}]}, **"val_classification_head_1_accuracy": {"direction": "min"**, "observations": [{"value": [0.7074671387672424], "step": 0}, {"value": [0.7387327551841736], "step": 1}, {"value": [0.7166140079498291], "step": 2}, {"value": [0.7433893084526062], "step": 3}, {"value": [0.7408947348594666], "step": 4}, {"value": [0.7753201127052307], "step": 5}, {"value": [0.7420588731765747], "step": 6}]}, **"val_classification_head_2_accuracy": {"direction": "max"**, "observations": [{"value": [0.48478296399116516], "step": 0}, {"value": [0.5732579231262207], "step": 1}, {"value": [0.5549642443656921], "step": 2}, {"value": [0.5700981020927429], "step": 3}, {"value": [0.5634458661079407], "step": 4}, {"value": [0.5606186389923096], "step": 5}, {"value": [0.5752536058425903], "step": 6}]}}}, "score": 23.471988677978516, "best_step": 1, "status": "COMPLETED"}

### Setup Details
Include the details about the versions of:
 - OS type and version:
 - Python: 3.6
 - autokeras: '1.0.12'
 - keras-tuner: '1.0.3'
 - numpy: '1.19.5'
 - tensorflow: '2.4.0'

How can I set all these directions to "max"? From one model I believe this is indeed altering predictions or should I just ignore this directions. The trial.json I'm referring to are located in each checkpoint.
